README.md

```diff
@@ -539,6 +543,96 @@ To avoid or work around caching, you can do the following:
 - Options.
 - Asynchronous by default (though can be run synchronously).
 
+## Usage for end users
+
+When configuring a tool, you can use multiple file formats and put these in multiple places.
+
+Usually, a tool would mention this in its own README file,
+but by default, these are the following places, where the `moduleName` variable represents the name of the tool:
+
+```js
+[
+  'package.json',
+  `.${moduleName}rc`,
+  `.${moduleName}rc.json`,
+  `.${moduleName}rc.yaml`,
+  `.${moduleName}rc.yml`,
+  `.${moduleName}rc.js`,
+  `.${moduleName}rc.cjs`,
+  `.config/${moduleName}rc`,
+  `.config/${moduleName}rc.json`,
+  `.config/${moduleName}rc.yaml`,
+  `.config/${moduleName}rc.yml`,
+  `.config/${moduleName}rc.js`,
+  `.config/${moduleName}rc.cjs`,
+  `${moduleName}.config.js`,
+  `${moduleName}.config.cjs`,
+]
+```
+
+The contents of these files are defined by the tool.
+For example, you can configure prettier to enforce semicolons at the end of the line
+using a file named `.config/prettierrc.yml`:
+
+```yaml
+semi: true
+```
+
+Additionally, you have the option to put a property named after the tool in your `package.json` file,
+with the contents of that property being the same as the file contents. To use the same example as above:
+
+```json
+{
+  "name": "your-project",
+  "dependencies": {},
+  "prettier": {
+    "semi": true
+  }
+}
+```
+
+This has the advantage that you can put the configuration of all tools
+(at least the ones that use cosmiconfig) in one file.
+
+You can also add a `cosmiconfig` key within your `package.json` file or create one of the following files
+to configure `cosmiconfig` itself:
+
+```
+.config.json
+.config.yaml
+.config.yml
+.config.js
+.config.cjs
+```
+
+The following properties are currently actively supported in these places:
+
+```yaml
+cosmiconfig:
+  # overrides where configuration files are being searched, this might be a minor performance optimization
+  # if all your configuration files use the same naming convention and format
+  searchPlaces:
+    - .config/{name}.yml
```

Comment:
This isn't one of the formats listed above; was it supposed to be `.config/{name}rc.yml`, or is this a separate search location?

Reply:
This is supposed to show that you can do whatever you want here.

Reply:
Ah, in that case it might be worthwhile to consider replacing the blurb about "minor performance optimization". I don't think end users will really care about a ms-level startup improvement.